py/frequenz/client/electricity_trading/_client.py

```diff
@@ -91,32 +101,78 @@ async def stream_gridpool_orders(  # pylint: disable=too-many-arguments
         Returns:
             Async generator of orders.
         """
-        gridpool_filter = GridpoolOrderFilter(
-            states=order_states,
+        gridpool_order_filter = GridpoolOrderFilter(
+            order_states=order_states,
             side=market_side,
             delivery_area=delivery_area,
             delivery_period=delivery_period,
             tag=tag,
         )
 
-        stream_key = (gridpool_id, gridpool_filter)
+        stream_key = (gridpool_id, gridpool_order_filter)
 
         if stream_key not in self._gridpool_orders_streams:
             self._gridpool_orders_streams[stream_key] = GrpcStreamingHelper(
                 f"electricity-trading-{stream_key}",
                 lambda: self._stub.ReceiveGridpoolOrdersStream(  # type: ignore
                     electricity_trading_pb2.ReceiveGridpoolOrdersStreamRequest(
                         gridpool_id=gridpool_id,
-                        filter=gridpool_filter.to_pb(),
+                        filter=gridpool_order_filter.to_pb(),
                     )
                 ),
                 lambda response: OrderDetail.from_pb(response.order_detail),
             )
         return self._gridpool_orders_streams[stream_key].new_receiver()
 
+    async def stream_gridpool_trades(  # pylint: disable=too-many-arguments
+        self,
+        gridpool_id: int,
+        trade_states: list[TradeState] | None = None,
+        trade_id_lists: list[int] | None = None,
+        market_side: MarketSide | None = None,
+        delivery_period: DeliveryPeriod | None = None,
+        delivery_area: DeliveryArea | None = None,
+    ) -> Receiver[Trade]:
+        """
+        Stream gridpool trades.
+
+        Args:
+            gridpool_id: ID of the gridpool to stream trades for.
+            trade_states: List of trade states to filter for.
+            trade_id_lists: List of trade IDs to filter for.
+            market_side: Market side to filter for.
+            delivery_period: Delivery period to filter for.
+            delivery_area: Delivery area to filter for.
+
+        Returns:
+            Async generator of trades.
```

Comment:
I'm not sure if `generator` is the right name. I'd suggest something around `the stream of gridpool trades` or `The gridpool trades stream(er)`.